src/Framework/IBuildEngine7.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// This interface extends <see cref="IBuildEngine6" /> to allow tasks to set whether they want to
+    /// log an error when a task returns without logging an error.
+    /// </summary>
+    public interface IBuildEngine7 : IBuildEngine6
+    {
+        public void SetEnableMSB4134(bool errorEnabled);
```

Comment:
Don't include the error code in the name of this method. Give it a descriptive name instead, like maybe
```suggestion
        public void AllowFailureWithoutError(bool errorEnabled);
```
?